darkmoon/api/v1/metadata/views.py

```diff
@@ -355,6 +362,51 @@ async def hash_comparison(
         sha256Hash = h_sha256.hexdigest()
         sha512Hash = h_sha512.hexdigest()
 
+        obj = MetadataEntity(
+            _id=PydanticObjectId(),
+            name=[inputFileName],
+            file_type=[inputFileType],
+            operating_system=[],
+            source_iso_name=[],
+            file_extension=[""],
+            hashes=Hashes(
+                md5=md5Hash,
+                sha1=sha1Hash,
+                sha256=sha256Hash,
+                sha512=sha512Hash,
+            ),
+            header_info=None,
+        )
+
+        # Check if hash is suspicious
+        sus_query = {
+            "$and": [
+                {
+                    "name": inputFileName,
+                },
+                {
+                    "file_type": inputFileType,
+                },
+            ],
+        }
+        susResults = await collection.find(sus_query).to_list(length=length)
+        susLi = [MetadataEntity.parse_obj(item) for item in susResults]
+        for metadata in susLi:
+            dbHashes = [
+                metadata.hashes.md5,
+                metadata.hashes.sha1,
+                metadata.hashes.sha256,
+                metadata.hashes.sha512,
+            ]
+            inputHashes = [md5Hash, sha1Hash, sha256Hash, sha512Hash]
+            if dbHashes != inputHashes:
+                await susCollection.insert_one(obj.dict())
+                response.status_code = status.HTTP_406_NOT_ACCEPTABLE
+                return UploadListMetadataEntityResponse(
+                    message="The hash of your file did not match the hash of the \
+existing, file in the database. File placed in suspicious collection.",
```

Comment:
This should probably be formatted a little cleaner to avoid the indentation change.